### ISSUE TITLE
YouTube URL should always be HTTPS

### ISSUE THREAD

**Issue by konklone** [digital-analytics-program/gov-wide-code#13]
The [call to the YouTube API](https://github.com/GSA/DAP-Gov-wide-GA-Code/blob/master/Universal-Federated-Analytics.1.0.js#L637) uses a protocol-relative URL on line 637:

```javascript
tag.src = "//www.youtube.com/iframe_api";
```

This should be updated to be:

```javascript
tag.src = "https://www.youtube.com/iframe_api";
```

This is the same issue as in #11, and in #8 -- it ensures that communication always happens over HTTPS, even when the website is using insecure HTTP. 

This prevents the site from being attacked by a network which rewrites or replaces the contents of `www.youtube.com/iframe_api`, which is trivial to do when accessed over `http://`. While it's certainly possible for a particularly hostile network to rewrite the contents of a .gov site using `http://`, domains like `www.youtube.com` are **far** more likely to be targeted for attack.

This is a user privacy and security issue that I hope the DAP will address in its next release.

**Comment by konklone** [comment]
A note that this is [still an issue in 2.0](https://github.com/GSA/DAP-Gov-wide-GA-Code/blob/master/Universal-Federated-Analytics.js#L730). The initial connection `www.youtube.com` is unprotected:

![screenshot from 2015-04-30 17 30 23](https://cloud.githubusercontent.com/assets/4592/7425043/a05a1174-ef5e-11e4-8064-a01bcfde5d24.png)

YouTube doesn't have a `forceSSL`-like function I'm aware of, but the initial connection can be protected by a 6-character fix: `https:`.

**Comment by konklone** [comment]
Is there any update on when this 6-character, all-upside fix might make it into DAP's deployed code?

**Comment by tdlowden** [comment]
Testing of the latest version of the code, which makes this fix, begins at the end of December. After assuring all runs well, the hosted code will be updated.

**Comment by konklone** [comment]
How'd testing go?

**Comment by tdlowden** [comment]
Well. Should be releasing by the end of the month.

**Comment by tdlowden** [comment]
Fixed with release of v3.1!
